### ISSUE TITLE
Java arguments specified via jvmArgs are ignored for incremental compilation

### ISSUE THREAD

**Issue by tashoyan** [davidB/scala-maven-plugin#507]
The original goal was to increase Java stack size for the Scala compilation with help of the `-Xss` Java argument:

```xml
<plugin>
  <groupId>net.alchim31.maven</groupId>
  <artifactId>scala-maven-plugin</artifactId>
  <configuration>
    ...
    <jvmArgs>
      <jvmArg>-Xss2m</jvmArg>
    </jvmArgs>
    ...
  </configuration>
<plugin>
```

It looks like all settings from `jvmArgs` are completely ignored for the goals `compile` and `testCompile`.
Surprisingly, `jvmArgs` are effective (not ignored) for the goal `doc-jar`.
The problem occurs for scala-maven-plugin versions starting from 4.0.0. For the version 3.4.6 the problem does not occur.

This is a severe problem, as it leaves no possibility to overcome StackOverflowError during Scala compilation. For example: [Stackoverflow in typechecker.Typers$Typer](https://github.com/scala/bug/issues/10604#issuecomment-376543621).

I am attaching a simple Maven project illustrating the problem. Here the JVM argument `-XshowSettings:all` is used as a marker: JVM settings are printed only when running the goal `doc-jar` - neither `compile` nor `testCompile` goals.
[scalac-xss.tar.gz](https://github.com/davidB/scala-maven-plugin/files/7062433/scalac-xss.tar.gz)
 
```text
java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

**Comment by davidB** [comment]
For `compile` and `testCompile`, `jvmArgs` are only used when `recompilationMode` is `all`. Since 4.5.0 the default value is `incremental`, in this case compilation (zinc based) is made in the same jvm than maven.

**Comment by tashoyan** [comment]
If we use the default value of `recompileMode`, then we have to tune the settings of the JVM running Maven. We can do it by putting a file `.mvn/jvm.config` in the project directory, as described in [Maven configuration](https://maven.apache.org/configure.html). This works!
Maybe makes sense to describe this behavior in the documentation. Otherwise, it is quite difficult to figure out, how to customize JVM settings for Scala compilation.

**Comment by slandelle** [comment]
I'm going to try and fix this.

**Comment by slandelle** [comment]
So actually, this is super complicated to implement. The incremental compiler needs a huge amount of parameters, it would be a huge effort. Not impossible, but not something I'll be able to tackle any time soon.

As a workaround, I think we could:

1. either disable the `incremental` mode when jvmArgs are defined
2. or default to `all` instead of `incremental` when jvmArgs are defined and crash if `incremental` is explicitly defined along with `jvmArgs`

@davidB WDYT?

**Comment by davidB** [comment]
I prefer 2. (less magic for end-user). I guess by "crash" you mean exit with an error message to explain the cause.

Thanks @slandelle for your support.

**Comment by slandelle** [comment]
@davidB Actually, I ended up implementing the process forking. PTAL #659 659

**Comment by slandelle** [comment]
Fixed in 4.8.0
